src/OrbitGl/ThreadTrack.cpp

```diff
@@ -282,22 +280,22 @@ void ThreadTrack::Draw(Batcher& batcher, TextRenderer& text_renderer,
   const float tracepoint_track_height = layout_->GetEventTrackHeightFromTid(thread_id_);
   const float track_width = size_[0];
 
+  DrawContext inner_draw_context = draw_context;
+  inner_draw_context.indentation_level = draw_context.indentation_level + 1;
```

Comment:
Another option is also to have a member function which does the job for you. Something like this:

```
struct DrawContext {
  // ...
  [[nodiscard]] DrawContext IncreasedIndentationLevel() const {
    auto copy = *this;
    copy.indentation_level++;
    return copy;
  }
};
```

I'm not saying this is the better approach. I'm just giving options. Feel free to pick whatever you like best.

Reply:
I thought about that, and also provide a method to update z_offset. Now that you are suggesting it, I will make them.